src/utilities.jl

```diff
@@ -25,36 +25,85 @@ function overlaps(ts::Vararg{Vector, N}) where {N}
     ret
 end
 
+"""
+    sorted_unique_merge(a, b) -> (c, idx_a, idx_b)
 
-function sorted_unique_merge(a::Vector, b::Vector)
-
-		i, na, j, nb = 1, length(a), 1, length(b)
-		c = similar(a, 0)
-
-		while (i <= na) && (j <= nb)
-				if a[i] < b[j]
-						push!(c, a[i])
-						i += 1
-				elseif a[i] > b[j]
-						push!(c, b[j])
-						j += 1
-				else
-						push!(c, a[i])
-						i += 1
-						j += 1
-				end #if
+Merge two arrays of sorted unique elements a and b to new array c,
+and calculate the indexes idx_a and idx_b mapping each element in a and b to their new locations in c.
+"""
+function sorted_unique_merge(::Type{IndexType}, a::Vector, b::Vector) where {IndexType}
+    # NOTE: Since merge is mostly load/store bound, the lower memory pressure from 32 bit indexing does help.
+    #       Would need to be reconsidered if datasets are larger, or branch the algorithm
+    i, na, j, nb = 1, length(a), 1, length(b)
+    c = similar(a, length(a) + length(b))
+    idx_a = Vector{IndexType}(length(a))
+    idx_b = Vector{IndexType}(length(b))
+    k = 1
+    @inbounds while (i <= na) && (j <= nb)
+		if a[i] < b[j]
```

Comment:
The indentation looks weird here. Please use space.

Reply:
Think the original code had some tabs and my editor was set to auto. Should be fixed.